### ISSUE TITLE
Route with 2 dynamic param separated with hyphens not working after upgrade from Remix 1.7.6

### ISSUE THREAD

**Issue by mikkpokk** [remix-run/remix#6731]
### What version of Remix are you using?

1.18.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

Create a route file /$slug-$id.tsx

```javascript
import {useParams} from '@remix-run/react'

export default () => {
    const params = useParams()

    return (
        <div>
            {JSON.stringify(params)}
        </div>
    )
}
```

### Expected Behavior

Route should match with paths 

/string-id (match a route /$slug-$id.tsx - `param.slug === 'string' && param.id === 'id'`)
/longer-string-id (match a route /$slug-$id.tsx -  `param.slug === 'longer-string' && param.id === 'id'`)
/a-bit-longer-string-44 (match a route /$slug-$id.tsx - `param.slug === 'a-bit-longer-string' && param.id === '44'`)

### Actual Behavior

/string-id - (Error: No route matches URL "/string-id")
/longer-string-id (Error: No route matches URL "/longer-string-id")
/a-bit-longer-string-44 (Error: No route matches URL "/a-bit-longer-string-44")

**Comment by kiliman** [comment]
Remix relies on React Router for routing. Dynamic params were *always* meant to require an **entire** URL segment. It was never documented to support multiple _sub_-params, prefixes, or any other type of pattern. There was a bug in an earlier version of RR that was fixed. That is why your upgrade no longer works the same.

You will need to update your routes to have a single param and then add code to parse the slug/id as required.
```ts
// routes/$slug.tsx

export function loader({ params }: LoaderArgs) {
  const { slug, id } = parseParam(params.slug)
  // ...
}

function parseParam(param: string) { 
  const i = param.lastIndexOf('-')
  // if param doesn't include hyphen: `throw new Response('Bad Request', { status: 400 })`
  const slug = param.substring(0, i)
  const id = param.substring(i + 1) 
  return { slug, id }; 
}
```

**Comment by brophdawg11** [comment]
This was a fix in [`React Router 6.5.0`](https://github.com/remix-run/react-router/releases/tag/react-router%406.5.0)

**Comment by mikkpokk** [comment]
Thank you for the clarification @kiliman.
